### PR TITLE
Restructure Binaries spec

### PR DESCRIPTION
**Pull request by janpio**
This PR suggests a bit of restructuring to the content of the Binaries spec to make it more approachable and understandable.

### CODE REVIEW

File: binaries/Readme.md [hunk 1]
```diff
@@ -53,15 +51,28 @@ Binaries are the artifacts generated by compiling Prisma's core (written in Rust
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
-## Motivation
+## Context
```
Comment (divyenduz) [inline binaries/Readme.md]: I personally feel that `Motivation` is a more approachable word than `Context`

Comment (janpio) [reply binaries/Readme.md]: "Motivation" explains why someone is doing something. In this case either why we have "Binaries" or why we have a spec about them. The text that follows on that headline does not do that, but gives context on binaries, right? (Alternative section headline could be "Introduction")

Comment (divyenduz) [reply binaries/Readme.md]: I prefer "introduction" over context but not strongly opinionated.

**Comment by janpio** [comment]
Merging although review from @timsuchanek is missing.

**Pull request merged**